hoss/dimension_utilities.py

```diff
@@ -73,38 +113,133 @@ def prefetch_dimension_variables(
 
     """
     required_dimensions = varinfo.get_required_dimensions(required_variables)
+    if len(required_dimensions) == 0:
```

Comment:
I think if a granule is CF compliant and has named dimensions, any variables that dont have the dimension attributes would be non-subsettable with that dimension. 
I dont see a situation where we would have to do a override for some of the variables

Reply:
There are definitely granules out there that are partially CF Convention compliant in some ways but not others. I think there is a better check here, rather than assuming presence of _some_ decent dimensions means a lack of anonymous dimensions.